cmdstanpy/cmdstan_args.py

```diff
@@ -0,0 +1,398 @@
+        adapt_delta: float = None,
+    ) -> None:
+        """Initialize object."""
+        self.warmup_iters = warmup_iters
+        self.sampling_iters = sampling_iters
+        self.save_warmup = save_warmup
+        self.thin = thin
+        self.max_treedepth = max_treedepth
+        self.metric = metric
+        self.metric_file = None
+        self.step_size = step_size
+        self.adapt_engaged = adapt_engaged
+        self.adapt_delta = adapt_delta
+
+    def validate(self, chains: int) -> None:
+        """
+        Check arguments correctness and consistency.
+
+        * adaptation and warmup args are consistent
+        * if file(s) for metric are supplied, check contents.
+        * length of per-chain lists equals specified # of chains
+        """
+        if type(chains) is not int or chains < 1:
+            raise ValueError(
+                "sampler expects number of chains to be greater than 0"
+            )
+
+        if self.warmup_iters is not None:
+            if self.warmup_iters < 0:
+                raise ValueError(
+                    'warmup_iters must be a non-negative integer'.format(
+                        self.warmup_iters
+                    )
+                )
+            if self.adapt_engaged and self.warmup_iters == 0:
+                raise ValueError(
+                    'adaptation requested but 0 warmup iterations specified, '
+                    'must run warmup iterations'
+                )
+
+        if self.sampling_iters is not None:
+            if self.sampling_iters < 0:
+                raise ValueError(
+                    'sampling_iters must be a non-negative integer'.format(
+                        self.sampling_iters
+                    )
+                )
+
+        if self.thin is not None:
+            if self.thin < 1:
+                raise ValueError(
+                    'thin must be at least 1, found {}'.format(self.thin)
+                )
+
+        if self.max_treedepth is not None:
+            if self.max_treedepth < 1:
+                raise ValueError(
+                    'max_treedepth must be at least 1, found {}'.format(
+                        self.max_treedepth
+                    )
+                )
+
+        if self.step_size is not None:
+            if isinstance(self.step_size, (int, float)):
+                if self.step_size < 0:
+                    raise ValueError(
+                        'step_size must be > 0, found {}'.format(self.step_size)
+                    )
+            else:
+                if len(self.step_size) != chains:
+                    raise ValueError(
+                        'number of step_sizes must match number of chains '
+                        ' found {} step_sizes for {} chains '.format(
+                            len(self.step_size), chains
+                        )
+                    )
+                for i in range(len(self.step_size)):
+                    if self.step_size[i] < 0:
```

Comment:
this block is better done with

    for step_size in self.step_size:
        if step_size < 0:

Reply:
fixed